docs/source/api/v4/sslkey_expirations.rst

```diff
@@ -0,0 +1,93 @@
+.. distributed under the License is distributed on an "AS IS" BASIS,
+.. WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+.. See the License for the specific language governing permissions and
+.. limitations under the License.
+..
+
+.. _to-api-sslkey_expirations:
+
+**********************
+``sslkey_expirations``
+**********************
+
+``GET``
+=======
+Retrieves SSL certificate expiration information.
+
+:Auth. Required: Yes
+:Roles Required: "admin"
+:Permissions Required: ACME:READ
+:Response Type:  Array
+
+Request Structure
+-----------------
+.. table:: Request Query Parameters
+
+	+-------------------+----------+--------------------------------------------------------------------------------------------------------+
+	| Name              | Required | Description                                                                                            |
+	+===================+==========+========================================================================================================+
+	| days              | no       | Return only the expiration information for SSL certificates expiring in the next given number of days. |
+	+-------------------+----------+--------------------------------------------------------------------------------------------------------+
+
+.. code-block:: http
+	:caption: Request Example
+
+	GET /api/4.0/sslkey_expirations?days=30 HTTP/1.1
+	Host: trafficops.infra.ciab.test
+	Accept: */*
+	Connection: keep-alive
+	Cookie: mojolicious=...
+
+Response Structure
+------------------
+:deliveryservice:    The :ref:`ds-xmlid` for the :term:`Delivery Service` corresponding to this SSL certificate.
+:deliveryservice_id: The ID for the :term:`Delivery Service` corresponding to this SSL certificate.
+:cdn:                The ID for the :abbr:`CDN (Content Delivery Network)` corresponding to this SSL certificate.
+:provider:           The provider of this SSL certificate, generally the name of the Certificate Authority or the :abbr:`ACME (Automatic Certificate Management Environment)` account.
+:expiration:         The expiration date of this SSL certificate.
+:federated:          A boolean indicating if this SSL certificate is use in a federated :term:`Delivery Service`.
+
+.. code-block:: http
+	:caption: Response Example
+
+	HTTP/1.1 200 OK
+	Access-Control-Allow-Credentials: true
+	Access-Control-Allow-Headers: Origin, X-Requested-With, Content-Type, Accept, Set-Cookie, Cookie
+	Access-Control-Allow-Methods: POST,GET,OPTIONS,PUT,DELETE
+	Access-Control-Allow-Origin: *
+	Content-Encoding: gzip
+	Content-Type: application/json
+	Permissions-Policy: interest-cohort=()
+	Set-Cookie: mojolicious=...; Path=/; Expires=Mon, 07 Jun 2021 22:52:20 GMT; Max-Age=3600; HttpOnly
+	Vary: Accept-Encoding
+	X-Server-Name: traffic_ops_golang/
+	Date: Mon, 07 Jun 2021 21:52:20 GMT
+	Content-Length: 384
+
+	{ "response": [
+		{
+			"deliveryservice": "foo1",
+			"deliveryservice_id": 1,
```

Comment:
Two things here:
1. I don't really think we need two ways to identify the Delivery Service here. I'd go with XMLID, both because it's easier to understand and because it has some bearing on the certs.

b) response payload parameters should be `camelCase` not `snake_case`

Reply:
so i originally was only going to return the XMLID but in TP it needs the id to navigate to the SSL Key page when the row gets clicked so i returned both so TP can show the XMLID but navigate to the page

Reply:
Well if that's really easier, then can't we just stick with `deliveryservice: number` and drop the XMLID? Alternatively, TP can just fetch DSes at the same time by just adding it to the `resolve` in the index.js and then use that to construct a map of XMLID to ID (which could also be done right in the `resolve` function) that can be used to construct the location on navigation, since that's a click handler and not a link it's really easy to do.

Reply:
👍